misk/src/main/kotlin/misk/MiskApplication.kt

```diff
@@ -88,6 +90,10 @@ class MiskApplication(private val modules: List<Module>, commands: List<MiskComm
     Runtime.getRuntime().addShutdownHook(object : Thread() {
       override fun run() {
         log.info { "received a shutdown hook! performing an orderly shutdown" }
+        if (injector.getExistingBinding(Key.get(WebConfig::class.java)) != null) {
```

Comment:
I think this would make a lot more sense at the start of `JettyService#shutDown()`. Saves having to check with the injector too, and makes the functionality testable (at least, in theory—testing a `sleep` call probably won't be nice).

Reply:
Unfortunately that does not actually help with the problem - the issue is with other services being shut down before the jetty service is finished servicing requests.

Reply:
I could move the sleep to the hibernate service instead, I suppose, at the risk of uncovering other services that may also be affected in the same way. Sleeping here avoids the risk of playing whack-a-mole for something that is intended to be a short-term workaround.